src/sources/dnstap/dns_message_parser.rs

```diff
@@ -0,0 +1,2055 @@
+            (index, buf)
+        } else {
+            let mut buf = Vec::<u8>::with_capacity(self.raw_message.len() * 2);
+            buf.extend(&self.raw_message);
+            buf.extend_from_slice(raw_rdata);
+            (self.raw_message.len(), buf)
+        };
+        self.raw_message_clone = Some(raw_message_clone_data);
+
+        BinDecoder::new(self.raw_message_clone.as_ref().unwrap()).clone(index as u16)
+    }
+
+    fn parse_wks_rdata(
+        &mut self,
+        raw_rdata: &[u8],
+    ) -> Result<(Option<String>, Option<Vec<u8>>), DnsMessageParserError> {
+        let mut decoder = BinDecoder::new(raw_rdata);
+        let address = parse_ipv4_address(&mut decoder)?;
+        let protocol = parse_u8(&mut decoder)?;
+        let port = {
+            let mut port_string = String::new();
+            let mut current_bit: u32 = 0;
+            while !decoder.is_empty() {
+                let mut current_byte = parse_u8(&mut decoder)?;
+                if current_byte == 0 {
+                    current_bit += 8;
+                    continue;
+                }
+                for _i in 0..8 {
+                    if current_byte & 0b1000_0000 == 0b1000_0000 {
+                        port_string.push_str(&current_bit.to_string());
+                        port_string.push(' ');
+                    }
+                    current_byte <<= 1;
+                    current_bit += 1;
+                }
+            }
+            port_string
+        };
+        Ok((
+            Some(format!("{} {} {}", address, protocol, port.trim_end())),
+            None,
+        ))
+    }
+
+    fn parse_a6_rdata(
+        &mut self,
+        raw_rdata: &[u8],
+    ) -> Result<(Option<String>, Option<Vec<u8>>), DnsMessageParserError> {
+        let mut decoder = BinDecoder::new(raw_rdata);
+        let prefix = parse_u8(&mut decoder)?;
+        let ipv6_address = {
+            let address_length = (128 - prefix) / 8;
+            let mut address_vec = parse_vec(&mut decoder, address_length)?;
+            if address_vec.len() < 16 {
+                let pad_len = 16 - address_length;
+                let mut padded_address_vec = vec![0; pad_len as usize];
+                padded_address_vec.extend(&address_vec);
+                address_vec = padded_address_vec;
+            }
+            let mut dec = BinDecoder::new(&address_vec);
+            parse_ipv6_address(&mut dec)?
+        };
+        let domain_name = parse_domain_name(&mut decoder)?;
+        Ok((
+            Some(format!(
+                "{} {} {}",
+                prefix,
+                ipv6_address,
+                domain_name.to_string()
+            )),
+            None,
+        ))
+    }
+
+    fn parse_loc_rdata(
+        &mut self,
+        raw_rdata: &[u8],
+    ) -> Result<(Option<String>, Option<Vec<u8>>), DnsMessageParserError> {
+        let mut decoder = BinDecoder::new(raw_rdata);
+        let _max_lattitude: u32 = 0x8000_0000 + 90 * 3_600_000;
```

Comment:
Nit: s/lattitude/latitude (mostly for the error message)

Reply:
Thanks, @lucperkins. It's been corrected.